src/main/java/core/basesyntax/strategy/operation/PurchaseOperation.java

```diff
@@ -0,0 +1,14 @@
+package core.basesyntax.strategy.operation;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+
+public class PurchaseOperation implements OperationHandler {
+    private StorageDao storageDao = new StorageDaoImpl();
+
+    @Override
+    public void operationType(String fruit, int amount) {
+        int newQuantity = storageDao.getFruit(fruit).getQuantity() - amount;
+        storageDao.update(fruit, newQuantity);
```

Comment:
You should check the result balance before updating the storage. If the new quantity is negative, you should throw a RuntimeException to prevent the storage from being updated with an invalid quantity.